### PR TITLE
Save lookdev prefab

### PR DESCRIPTION
**Pull request by JT00y**
## References
closes #_insert number here_

### CODE REVIEW

File: packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx [hunk 1]
```diff
@@ -68,99 +74,168 @@ export default function CreatePrefabPanel({ entity }: { entity: Entity }) {
...
-        getMutableState(SelectionState).selectedEntities.set([])
-        getComponent(entity, TransformComponent).matrix.identity()
-        await exportRelativeGLTF(prefabEntity, srcProject, fileName)
-
-        const resources = await API.instance.service(staticResourcePath).find({
-          query: { key: 'projects/' + srcProject + '/' + fileName }
-        })
-        if (resources.data.length === 0) {
-          throw new Error('User not found')
+        if (isExportLookDev) {
```
Comment (HexaField) [inline packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx:224]: we definitely need to break this down into smaller functions

File: packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx [hunk 2]
```diff
@@ -68,99 +74,168 @@ export default function CreatePrefabPanel({ entity }: { entity: Entity }) {
...
+          const lookDevComponent: Component[] = [
+            SkyboxComponent,
+            HemisphereLightComponent,
+            DirectionalLightComponent,
+            PostProcessingComponent
+          ]
+          const prefabEntity = createEntity()
+          const obj = new Scene()
+          addObjectToGroup(prefabEntity, obj)
+          proxifyParentChildRelationships(obj)
```
Comment (dinomut1) [inline packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx]: don't need this

File: packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx [hunk 3]
```diff
@@ -68,99 +74,168 @@ export default function CreatePrefabPanel({ entity }: { entity: Entity }) {
...
+          const rootEntity = getState(EditorState).rootEntity
+          iterateEntityNode(rootEntity, (entity) => {
+            lookDevComponent.forEach((component) => {
+              if (hasComponent(entity, component)) {
+                lookdevEntity.push(entity)
+              }
+            })
+          })
```
Comment (dinomut1) [inline packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx]: this is nice. Add a check if lookdevEntity already has this component somewhere.

File: packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx [hunk 4]
```diff
@@ -68,99 +74,168 @@ export default function CreatePrefabPanel({ entity }: { entity: Entity }) {
...
+          await EditorControlFunctions.removeObject(lookdevEntity)
+          lookdevEntity.forEach((entity) => {
+            removeEntity(entity)
+          })
+          addMediaNode(fileURL)
+            .catch((error) => getMutableState(EditorErrorState).error.set(error))
+            .then((uuid) => {
+              const entity = UUIDComponent.getEntityByUUID(uuid)
+              setComponent(entity, NameComponent, prefabName.value)
+            })
```
Comment (dinomut1) [inline packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx]: remove this

File: packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx [hunk 5]
```diff
@@ -68,99 +74,168 @@ export default function CreatePrefabPanel({ entity }: { entity: Entity }) {
...
+          PopoverState.hidePopupover()
+          defaultPrefabFolder.set('assets/custom-prefabs')
+          prefabName.set('prefab')
+          prefabTag.set([])
+          isOverwriteModalVisible.set(false)
+          isOverwriteConfirmed.set(false)
+          PopoverState.showPopupover(<PrefabConfirmationPanelDialog entity={entity as Entity} />)
```
Comment (dinomut1) [inline packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx]: this works, but put it in a helper function that you call at the end of both conditionals

**Review by dinomut1 — CHANGES_REQUESTED** [review]

**Review by dinomut1 — APPROVED** [review]
alright!

**Pull request merged**